sdk/appconfiguration/azure-data-appconfiguration/swagger/src/main/java/AppConfigCustomization.java

```diff
@@ -0,0 +1,146 @@
+        classCustomization.getMethod("getLabel")
+            .getJavadoc()
+            .setDescription("Get the label property: Filters {@link ConfigurationSetting} by their label field.");
+
+        classCustomization = classCustomization.customizeAst(
+            ast -> ast.getClassByName("KeyValueFilter")
+                       .get()
+                       .getDefaultConstructor().get()
+                       .remove())
+                                 .removeMethod("setKey")
+                                 .rename("SnapshotSettingFilter");
+
+        classCustomization.addConstructor(joinWithNewline(
+            "public SnapshotSettingFilter(String key) {",
+            "   this.key = key;",
+            "}"))
+            .getJavadoc()
+            .setDescription("Constructor of {@link SnapshotSettingFilter}.")
+            .setParam("key", "The key value to set.");
+    }
+
+    private void customizeKeyValueFields(ClassCustomization classCustomization) {
+        classCustomization.addImports("java.util.Locale;");
+        classCustomization.addImports("com.azure.data.appconfiguration.ConfigurationAsyncClient;");
+
+        classCustomization = classCustomization.removeMethod("values");
+
+        classCustomization.getConstant("KEY")
+            .getJavadoc()
+            .setDescription("Populates the {@link ConfigurationSetting#getKey()} from the service.");
+
+        classCustomization.getConstant("LABEL")
+            .getJavadoc()
+            .setDescription("Populates the {@link ConfigurationSetting#getLabel()} from the service.");
+
+        classCustomization.getConstant("VALUE")
+            .getJavadoc()
+            .setDescription("Populates the {@link ConfigurationSetting#getValue()} from the service.");
+
+        classCustomization.getConstant("CONTENT_TYPE")
+            .getJavadoc()
+            .setDescription("Populates the {@link ConfigurationSetting#getContentType()} from the service.");
+
+        classCustomization.getConstant("LAST_MODIFIED")
+            .getJavadoc()
+            .setDescription("Populates the {@link ConfigurationSetting#getLastModified()} from the service.");
+
+        classCustomization.getConstant("TAGS")
+            .getJavadoc()
+            .setDescription("Populates the {@link ConfigurationSetting#getTags()} from the service.");
+
+        classCustomization.getConstant("ETAG")
+            .getJavadoc()
+            .setDescription("Populates the {@link ConfigurationSetting#getETag()} from the service.");
+
+        MethodCustomization fromString = classCustomization.getMethod("fromString");
+        fromString.getJavadoc()
+            .setDescription("Creates or finds a {@link SettingFields} from its string representation.")
+            .setReturn("the corresponding {@link SettingFields}");
+        fromString.removeAnnotation("JsonCreator");
+
+        classCustomization = classCustomization.customizeAst(
```

Comment:
If the renaming works, keep this constructor as it should be included. Same goes for keeping the `values` method as well.